### ISSUE TITLE
Fallback icon

### ISSUE THREAD

**Issue by boy51** [iconify-design/iconify-react#22]
When using this component without api, it would be nice to provide fallback icons in case the icon is not in the bundle.

Right now the only way I see is manually checking via a ref and then loading a fallback.

**Comment by cyberalien** [comment]
In component with API it would make sense because data might take a while to load.

However, if you are using component without API, you are passing icon data as parameter. That means you can see if data is available and render whatever else you want to render.

**Comment by boy51** [comment]
Unless I'm bundling my Icons with `addCollection` and then use strings to render.

```
import { Icon } from "@iconify/react";

<Icon icon="mdi:delete" />
```

Obviously the assumption is all icons are available in the bundle. But until someone manages to setup their CI to bundle icons automatically, it would be very useful to have props to be able to provide fallbacks.

**Comment by cyberalien** [comment]
That's a very good example. 

Few possible solutions I can think of:

1. Exposing storage, so in your use case you can wrap component in a function that checks storage and renders placeholder if icon data is not in storage. I think this is the simplest solution that doesn't change behavior and doesn't add to bundle size.

2. Using something like empty string as default icon, so it can be added using `addIcon('', data)`. Drawback is that every icon will have the same fallback icon, which might not be ideal.

3. Adding extra parameter for fallback icon, like `<Icon icon="mdi:delete" fallback="jam:delete" />`. However that increases component size a bit and what would happen if fallback icon is not bundled?

4. Rendering child nodes when icon data is not available, like `<Icon icon="mdi:delete">❌</Icon>`. This seems like a simple solution, however, I'm not sure if there will be a performance hit. Do you know if React parses children even if they are not used?

**Comment by boy51** [comment]
I think these are all decent solutions. I wouldn't prefer the last one. I think its rather confusing syntax.

I realized there is a way to do this right now by using an enum.

```ts
enum AppIcons {
   Delete = "mdi:delete"
}
```

The icon bundling script could bundle Icons from the enum using `Object.values(AppIcons)`. Then we can use the enum to render the icon and you will know the icon is bundled.

This doesn't work in all situations of course, so some sort of fallback solutions would still be great.

Another way to do this would be an `onError` prop. This would make it pretty easy to wrap this component to do exactly what you want.

**Comment by cyberalien** [comment]
I've tried implementing last option, which shows child nodes if icon is missing, but it got a bit messy. Implementing it is simple, problems arise when mixing it with forwarding refs and with child component that relies on API.

To fix that mess I've decided to completely restructure component, changing order of logic. This is a large task, which will take a while.

**Comment by cyberalien** [comment]
Also fixed in 2.0.0-rc.9 of `@iconify/react` package, so you could use it today, but it gets a bit sketchy when merged into `@iconify/react-with-api`, so `@iconify/react-with-api` will not include that update.

**Comment by cyberalien** [comment]
Properly fixed in API package in 3.0.0-dev.

Unfortunately due to mess with references it cannot be fixed in earlier version. 3.0.0 has been rewritten from ground up to account for mistakes made during development of previous version.
